devops/viime/R/analyses.R

```diff
@@ -152,3 +152,46 @@ clustered_heatmap <- function(measurements) {
 
   list(data=OUT_CSV, row=unclass(ddr), col=unclass(ddc))
 }
+
+#' roc_analysis
+#'
+#' @export
+roc_analysis <- function(measurements, groups, group_name, column_name, method) {
+  library(pROC)
+  library(randomForest)
+
+  df = read.csv(measurements, row.names=1, check.names=FALSE)
+  groups = read.csv(groups, row.names=1, check.names=FALSE)
+  groups = as.factor(groups[, 1])
+  group_mask = ifelse(groups == group_name, 1, 0)
+
+  #############
+  #Get predicted values
+
+  column = df[[ column_name ]];
+
+  if (method == "logistic_regression") {
+    #-# Logistic Regression
+    glm.fit = glm(group_mask ~ column, family=binomial)
+
+    # Save predicted values from logistic regression
+    pred <- glm.fit$fitted.values
+  } else if (method == "random_forest") {
+    #-# Random forest
+    rf.model <- randomForest::randomForest(factor(group_mask)~column)
+
+    # Save predicted values from random forest
+    pred <- rf.model$votes[,1]
+  } else {
+    stop("Invalid method")
+  }
+
+  #############
+  #ROC analysis
+  roc.a <- pROC::roc(group_mask, pred, plot=FALSE)
```

Comment:
The web client also needs the area under the curve, which I believe can be calculated with `auc(roc.a)`. (https://rdrr.io/cran/pROC/man/auc.html)

Reply:
I added it in [`530e90e` (#583)](https://github.com/girder/viime/pull/583/commits/530e90e9dcf0a9108432bb47b1de4dfbc03418a8).  The R server won't let you return heterogeneous types, so it returns the scalar value repeated in a list.